### PR TITLE
Develop - All pending features for Milestone

### PR DESCRIPTION
**Pull request by anandjainjk**

### CODE REVIEW

File: src/main/resources/project/drivers/BerksCookbookDriver.pl [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    my $cookbook_name =
+      ( $ec->getProperty("cookbook_name") )->findvalue('//value')->string_value;
+    my $additional_options =
+      ( $ec->getProperty("additional_options") )->findvalue('//value')
+      ->string_value;
+
+    $ec->abortOnError(1);
+
+    #Variable that stores the command to be executed
+    my $command = " yes | berks cookbook ";
```
Comment (nikhilv) [inline src/main/resources/project/drivers/BerksCookbookDriver.pl]: Does this work on Windows?

File: src/main/resources/project/drivers/BerksCookbookDriver.pl [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    if ( $additional_options && $additional_options ne '' ) {
+
+        $command = $command . " " . $additional_options;
+    }
+
+    #Print out the command to be executed
+    print "\nCommand to be executed: \n$command \n\n";
+
+    #Executes the command
+    system("$command");
```
Comment (nikhilv) [inline src/main/resources/project/drivers/BerksCookbookDriver.pl]: Checking of exit codes

File: src/main/resources/project/drivers/BerksInitDriver.pl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      ( $ec->getProperty("additional_options") )->findvalue('//value')
+      ->string_value;
+
+    $ec->abortOnError(1);
+
+    my $filename = "Berksfile";
+    my $filePath = "";
+
+    #Variable that stores the command to be executed
+    my $command = " yes | berks init ";
```
Comment (nikhilv) [inline src/main/resources/project/drivers/BerksInitDriver.pl]: Does this work on Windows?

File: src/main/resources/project/drivers/EditNodeDriver.pl [hunk 4]
```diff
@@ -57,27 +59,34 @@ sub main {
...
+    $ec->abortOnError(1);  
+
     #Write to file
-    my $file = "/tmp/nodedata.json";
-    if ( $::g_node_data && $::g_node_data ne '' ) {
-        open my $fh, '>', $file or die "can't open $file: $!";
-        print $fh $::g_node_data;
+    my $dir = cwd();
+    my $fh = tempfile( );
+    my $template = "nodedataXXXX";
```
Comment (nikhilv) [inline src/main/resources/project/drivers/EditNodeDriver.pl]: Nice! This looks better

**Comment by nikhilv** [comment]
We need to reduce batch size. It's difficult to really review all the changes that are going on here. I reviewed the previous issues I filed and found that some things were not addressed such as https://github.com/electric-cloud/EC-Chef/issues/19 (exit codes) and still see some linux specific behavior.

File: src/main/resources/project/drivers/BerksInitDriver.pl (hunk 3 above)
Comment (chanijindal1) [inline src/main/resources/project/drivers/BerksInitDriver.pl]: Hi,
There's a bug in berkshelf due to which it gives overwrite prompts to users
. Hence we piped yes with every berks command .
The same could not be done on windows .
However , this problem can be solved by using the Expect script which
requires  it's library (
http://search.cpan.org/~rgiersig/Expect-1.15/Expect.pod).

Than you,
Chani Jindal

**Pull request merged**